web/src/components/tasks/cigarette-license/CigaretteLicenseReview.tsx

```diff
@@ -0,0 +1,411 @@
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidMailingAddressZipCode } = useFormContextFieldHelpers(
+    "mailingAddressZipCode",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidSalesInfoStartDate } = useFormContextFieldHelpers(
+    "salesInfoStartDate",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidSalesInfoSupplier } = useFormContextFieldHelpers(
+    "salesInfoSupplier",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidContactName } = useFormContextFieldHelpers(
+    "contactName",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidContactPhoneNumber } = useFormContextFieldHelpers(
+    "contactPhoneNumber",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidContactEmail } = useFormContextFieldHelpers(
+    "contactEmail",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidSignature } = useFormContextFieldHelpers(
+    "signature",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidSignerRelationship } = useFormContextFieldHelpers(
+    "signerRelationship",
+    DataFormErrorMapContext,
+  );
+  const { setIsValid: setIsValidSignerName } = useFormContextFieldHelpers(
+    "signerName",
+    DataFormErrorMapContext,
+  );
+
+  const validateAllFields = (): boolean => {
+    let isValid = true;
+
+    if (isTradeNameLegalStructureApplicable(profileData.legalStructureId)) {
+      const isResponsibleOwnerNameValid = cigaretteLicenseData.responsibleOwnerName !== "";
+      const isTradeNameValid = cigaretteLicenseData.tradeName !== "";
+      setIsValidResponsibleOwnerName(isResponsibleOwnerNameValid);
+      setIsValidTradeName(isTradeNameValid);
+      if (!isResponsibleOwnerNameValid || !isTradeNameValid) isValid = false;
+    } else {
+      const isBusinessNameValid = cigaretteLicenseData.businessName !== "";
+      setIsValidBusinessName(isBusinessNameValid);
+      if (!isBusinessNameValid) isValid = false;
+    }
+
+    const validations = [
+      { field: cigaretteLicenseData.taxId, setter: setIsValidTaxId },
+      { field: cigaretteLicenseData.addressLine1, setter: setIsValidAddressLine1 },
+      { field: cigaretteLicenseData.addressCity, setter: setIsValidAddressCity },
+      { field: cigaretteLicenseData.addressState, setter: setIsValidAddressState },
+      { field: cigaretteLicenseData.addressZipCode, setter: setIsValidAddressZipCode },
+      { field: cigaretteLicenseData.mailingAddressLine1, setter: setIsValidMailingAddressLine1 },
+      { field: cigaretteLicenseData.mailingAddressCity, setter: setIsValidMailingAddressCity },
+      { field: cigaretteLicenseData.mailingAddressState, setter: setIsValidMailingAddressState },
+      {
+        field: cigaretteLicenseData.mailingAddressZipCode,
+        setter: setIsValidMailingAddressZipCode,
+      },
+      { field: cigaretteLicenseData.salesInfoStartDate, setter: setIsValidSalesInfoStartDate },
```

Comment:
For reviewers: Depending on whether or not Step 3 has been merged (and this subsequently rebased), you may not be able to complete the form through the application UI. You can temporarily comment out these lines for Sales Info Start Date and Sales Info Supplier in order to test the submission functionality in the meantime.